Date: Mon, 20 Mar 2023 02:39:35 +0000
Subject: msr: disable MSR writes by default

Suggested by Benjamin Herrenschmidt <benh@amazon.com>, increases security
by disallowing user-space to change CPU behavior.

MSR writes can be reenabled by doing one of the following:

o Writing "on" or "default" to /sys/module/msr/parameters/allow_writes
o Booting with msr.allow_writes=<on|default>

Signed-off-by: Luiz Capitulino <luizcap@amazon.com>

diff --git a/arch/x86/kernel/msr.c b/arch/x86/kernel/msr.c
--- a/arch/x86/kernel/msr.c
+++ b/arch/x86/kernel/msr.c
@@ -48,7 +48,7 @@ enum allow_write_msrs {
 	MSR_WRITES_DEFAULT,
 };
 
-static enum allow_write_msrs allow_writes = MSR_WRITES_DEFAULT;
+static enum allow_write_msrs allow_writes = MSR_WRITES_OFF;
 
 static ssize_t msr_read(struct file *file, char __user *buf,
 			size_t count, loff_t *ppos)
